pkg/podexecutor/staticpod_windows.go

```diff
@@ -0,0 +1,345 @@
+	}
+
+	args = append(getArgsList(extraArgs), args...)
+	logrus.Infof("Running RKE2 kubelet %s", args)
+	go func() {
+		for {
+			cmd := exec.Command(s.KubeletPath, args...)
+			cmd.Stdout = os.Stdout
+			cmd.Stderr = os.Stderr
+			addDeathSig(cmd)
+
+			err := cmd.Run()
+			logrus.Errorf("Kubelet exited: %v", err)
+
+			time.Sleep(5 * time.Second)
+		}
+	}()
+
+	if err := s.cni.Start(s.cniConig); err != nil {
+		return err
+	}
+	return nil
+}
+
+// KubeProxy starts the kubeproxy in a subprocess with watching goroutine.
+func (s *StaticPodConfig) KubeProxy(args []string) error {
+	extraArgs := map[string]string{
+		"hostname-override": s.cniConig.CalicoConfig.Hostname,
+		"v":                 "4",
+		"proxy-mode":        "kernelspace",
+		"kubeconfig":        s.cniConig.NodeConfig.AgentConfig.KubeConfigKubeProxy,
+		"network-name":      s.cniConig.NetworkName,
+		"bind-address":      s.cniConig.BindAddress,
+	}
+
+	if isDsrSupported() {
+		logrus.Infof("WinDSR support is enabled")
+		extraArgs["feature-gates"] = addFeatureGate(extraArgs["feature-gates"], "WinDSR=true")
+		extraArgs["enable-dsr"] = "true"
+	}
+
+	for range time.Tick(time.Second * 5) {
+		// TODO: Pull from CNI Config
+		endpoint, err := hcsshim.GetHNSEndpointByName("Calico_ep")
+		if err != nil {
+			logrus.WithError(err).Warning("can't find %s, retrying", "Calico_ep")
+			continue
+		}
+		extraArgs["source-vip"] = endpoint.IPAddress.String()
+		break
+	}
+
+	logrus.Infof("Deleting HNS policies before kube proxy starts.")
+	policies, _ := hcsshim.HNSListPolicyListRequest()
+	for _, policy := range policies {
+		policy.Delete()
+	}
+
+	args = append(getArgsList(extraArgs), args...)
+	logrus.Infof("Running RKE2 Kubeproxy %s", args)
+	go func() {
+		for {
+			cmd := exec.Command("C:/var/lib/rancher/rke2/bin/kube-proxy.exe", args...)
+			cmd.Stdout = os.Stdout
+			cmd.Stderr = os.Stderr
+			addDeathSig(cmd)
+			err := cmd.Run()
+			logrus.Errorf("Kubeproxy exited: %v", err)
+			time.Sleep(5 * time.Second)
+		}
+	}()
+
+	return nil
+}
+
+// APIServer sets up the apiserver static pod once etcd is available, returning the authenticator and request handler.
+func (s *StaticPodConfig) APIServer(ctx context.Context, etcdReady <-chan struct{}, args []string) (authenticator.Request, http.Handler, error) {
+	panic("api server is unsupported on windows")
+}
+
+// Scheduler starts the kube-scheduler static pod, once the apiserver is available.
+func (s *StaticPodConfig) Scheduler(apiReady <-chan struct{}, args []string) error {
+	panic("scheduler is unsupported on windows")
+}
+
+// onlyExisting filters out paths from the list that cannot be accessed
+func onlyExisting(paths []string) []string {
+	existing := []string{}
+	for _, path := range paths {
+		if _, err := os.Stat(path); err == nil {
+			existing = append(existing, path)
+		}
+	}
+	return existing
+}
+
+// after calls a function after a message is received from a channel.
+func after(after <-chan struct{}, f func() error) error {
```

Comment:
Do we need to return an error? Doesn't look like it could ever happen.

Reply:
Those are unused and I will remove those. This is the same in the linux version.